OpenKh.Engine.MonoGame/KingdomTexture.cs

```diff
@@ -40,6 +43,30 @@ public void Dispose()
         }
     }
 
+    public class Tim2KingdomTexture : IKingdomTexture
+    {
+        private static readonly Vector2 DefaultRegion = new Vector2(0, 1);
+
+        public Tim2KingdomTexture(byte[] texture, GraphicsDevice graphics)
+        {
+            List<Tm2> tm2 = Tm2.Read(new MemoryStream(texture)).ToList();
```

Comment:
To squeeze a little amount of performance, as you already know that the returned value from `Tm2.Read` will always have a single element, I suggest you to do `Tm2.Read(new MemoryStream(texture)).First()`.

Reply:
Made it even shorter, like this:
```
public Tim2KingdomTexture(byte[] texture, GraphicsDevice graphics)
{
   Texture2D = Tm2.Read(new MemoryStream(texture)).First().CreateTexture(graphics);
}
```